### PR TITLE
Add REST APIs for subscription blocking

### PR DESCRIPTION
**Pull request by ruwiniwj**
This PR adds REST APIs for the following

- get all subscription of an API 
- get details of a specific subscription
- blocking a subscription
- unblocking a subscription

Fix: https://github.com/wso2/product-apim/issues/5253

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 1]
```diff
@@ -6644,6 +6644,41 @@ definitions:
...
+      applicationId:
+        type: string
+        example: 01234567-0123-0123-0123-012345678901
+      name:
+        type: string
+        example: CalculatorApp
+      subscriber:
+        type: string
+        example: admin
+      throttlingPolicy:
```
Comment (tmkasun) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: Shall we remove this property?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 2]
```diff
@@ -6644,6 +6644,41 @@ definitions:
...
+        type: string
+        example: Unlimited
+      description:
+        type: string
+        example: Sample calculator application
+      status:
```
Comment (tmkasun) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: Shall we remove this property?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 3]
```diff
@@ -6644,6 +6644,41 @@ definitions:
...
+        type: string
+        example: APPROVED
+        default: ""
+      groups:
```
Comment (tmkasun) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: Shall we remove this property?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 4]
```diff
@@ -6644,6 +6644,41 @@ definitions:
...
+        type: array
+        items:
+          type: string
+        example: ""
+      subscriptionCount:
+        type: integer
+      attributes:
```
Comment (tmkasun) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: Shall we remove this property?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/utils/mappings/SubscriptionMappingUtil.java [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+    }
+
+    /**
+     * Convert APIIdentifier to an APIInfoDTO
+     *
+     * @param apiId the API identifier to be converted
+     * @return APIInfoDTO corresponding to the APIIdentifier
+     * @throws APIManagementException If an error occurs when getting logged in provider or when getting lightweight API
+     */
+    private static APIInfoDTO FromAPIIdToAPIinfoDTO(APIIdentifier apiId) throws APIManagementException{
```
Comment (tmkasun) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/utils/mappings/SubscriptionMappingUtil.java]:
```suggestion
private static APIInfoDTO FromAPIIdToAPIInfoDTO(APIIdentifier apiId) throws APIManagementException{
```

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/utils/mappings/SubscriptionMappingUtil.java [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+     * Converts a SubscribedAPI object into SubscriptionDTO
+     *
+     * @param subscription SubscribedAPI object
+     * @return SubscriptionDTO corresponds to SubscribedAPI object
+     */
+    public static SubscriptionDTO fromSubscriptionToDTO(SubscribedAPI subscription)
+            throws APIManagementException {
+        SubscriptionDTO subscriptionDTO = new SubscriptionDTO();
+        subscriptionDTO.setSubscriptionId(subscription.getUUID());
+        subscriptionDTO.setApiInfo(FromAPIIdToAPIinfoDTO(subscription.getApiId()));
```
Comment (tmkasun) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/utils/mappings/SubscriptionMappingUtil.java]: shall we remove this line?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml (hunk 1 above)
Comment (ruwiniwj) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: Fixed with https://github.com/wso2/carbon-apimgt/pull/6772

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml (hunk 2 above)
Comment (ruwiniwj) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: Fixed with https://github.com/wso2/carbon-apimgt/pull/6772

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml (hunk 3 above)
Comment (ruwiniwj) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: Fixed with https://github.com/wso2/carbon-apimgt/pull/6772

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml (hunk 4 above)
Comment (ruwiniwj) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: Fixed with https://github.com/wso2/carbon-apimgt/pull/6772

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/utils/mappings/SubscriptionMappingUtil.java (hunk 5 above)
Comment (ruwiniwj) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/utils/mappings/SubscriptionMappingUtil.java]: Removed with https://github.com/wso2/carbon-apimgt/pull/6772

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/utils/mappings/SubscriptionMappingUtil.java (hunk 6 above)
Comment (ruwiniwj) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/utils/mappings/SubscriptionMappingUtil.java]: Fixed with https://github.com/wso2/carbon-apimgt/pull/6772

**Pull request merged**